paths_cli/compiling/core.py

```diff
@@ -0,0 +1,300 @@
+import json
+import yaml
+
+from collections import namedtuple
+
+import logging
+
+from .errors import InputError
+from paths_cli.utils import import_thing
+from paths_cli.plugin_management import OPSPlugin
+
+def listify(obj):
+    listified = False
+    if not isinstance(obj, list):
+        obj = [obj]
+        listified = True
+    return obj, listified
+
+def unlistify(obj, listified):
+    if listified:
+        assert len(obj) == 1
+        obj = obj[0]
+    return obj
+
+REQUIRED_PARAMETER = object()
+
+class Parameter:
+    SCHEMA = "http://openpathsampling.org/schemas/sim-setup/draft01.json"
+    def __init__(self, name, loader, *, json_type=None, description=None,
+                 default=REQUIRED_PARAMETER, aliases=None):
+        if isinstance(json_type, str):
+            try:
+                json_type = json.loads(json_type)
+            except json.decoder.JSONDecodeError:
+                # TODO: maybe log this in case it represents an issue?
+                pass
+
+        self.name = name
+        self.loader = loader
+        self._json_type = json_type
+        self._description = description
+        self.default = default
+
+    @property
+    def required(self):
+        return self.default is REQUIRED_PARAMETER
+
+    @property
+    def json_type(self):
+        return self._get_from_loader(self.loader, 'json_type',
+                                     self._json_type)
+
+    @property
+    def description(self):
+        return self._get_from_loader(self.loader, 'description',
+                                     self._description)
+
+    @staticmethod
+    def _get_from_loader(loader, attr_name, attr):
+        if attr is None:
+            try:
+                attr = getattr(loader, attr_name)
+            except AttributeError:
+                pass
```

Comment:
this can be unwrapped by using the `default` on `getattr`:
```suggestion
            attr = getattr(loader, attr_name, None)
```